src/Oscoin/Storage/ContentStore.hs

```diff
@@ -0,0 +1,54 @@
+module Oscoin.Storage.ContentStore
+    ( ContentStore
+    , storeContent
+    , lookupContent
+
+    , newIoContentStore
+    , mkStateContentStore
+    , hoistContentStore
+    ) where
+
+import           Oscoin.Prelude
+
+import qualified Oscoin.Crypto.Hash as Crypto
+
+import qualified Data.Map as Map
+import           Lens.Micro
+import           Lens.Micro.Mtl
+
+data ContentStore crypto a m = ContentStore
+    { storeContent  :: a -> m ()
+    , lookupContent :: Crypto.Hashed crypto a -> m (Maybe a)
+    }
+
+hoistContentStore :: (forall x. m x -> n x) -> ContentStore crypto a m -> ContentStore crypto a n
+hoistContentStore lift' c = ContentStore
```

Comment:
category theory nitpick: this is technically a natural transformation, but I can live with `lift'` 😬